packages/wallet/ui/src/components/ProvisionDialog.jsx

```diff
@@ -0,0 +1,123 @@
+import CircularProgress from '@mui/material/CircularProgress';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import Link from '@mui/material/Link';
+import { useState } from 'react';
+
+import { withApplicationContext } from '../contexts/Application';
+
+const steps = {
+  INITIAL: 0,
+  AWAITING_APPROVAL: 1,
+  IN_PROGRESS: 2,
+};
+
+const errors = {
+  NO_SIGNER: 'Cannot sign a transaction in read only mode, connect to keplr.',
+};
+
+const ProvisionDialog = ({ onClose, open, address, href, keplrConnection }) => {
+  const [currentStep, setCurrentStep] = useState(steps.INITIAL);
+  const [error, setError] = useState('');
+
+  const provisionWallet = async () => {
+    const {
+      signers: { interactiveSigner },
+    } = keplrConnection;
+    if (!interactiveSigner) {
+      setError(errors.NO_SIGNER);
+      return;
+    }
+
+    setError('');
+    setCurrentStep(steps.AWAITING_APPROVAL);
+
+    try {
+      await interactiveSigner.submitProvision();
+    } catch (e) {
+      setCurrentStep(steps.INITIAL);
+      setError(e.message);
+      return;
+    }
+
+    setCurrentStep(steps.IN_PROGRESS);
+  };
+
+  const progressIndicator = text => (
+    <Box>
+      <Box
+        sx={{
+          margin: 'auto',
+          display: 'flex',
+          flexDirection: 'row',
+          justifyContent: 'center',
+        }}
+      >
+        <CircularProgress />
+      </Box>
+      <DialogContentText sx={{ pt: 2 }}>{text}</DialogContentText>
+    </Box>
+  );
+
+  const content = (() => {
+    switch (currentStep) {
+      case steps.INITIAL:
+        return (
+          <div>
+            <DialogContentText>
+              <b>Network Config</b>:{' '}
+              <Link href={href} underline="none" color="rgb(0, 176, 255)">
+                {href}
+              </Link>
+            </DialogContentText>
+            <DialogContentText sx={{ pt: 2 }}>
+              <b>Wallet Address:</b> {address}
+            </DialogContentText>
+            <DialogContentText sx={{ pt: 2 }}>
+              There is no smart wallet provisioned for this address yet. A fee
```

Comment:
if we're not going to read it from params, at least make it a const. should have a comment indicating the tech debt and why

Reply:
Made a const and TODO